gh-actions/go/code-sanity/action.yaml

```diff
@@ -25,66 +25,102 @@ runs:
+      shell: bash
+    - name: Get tooling version
+      shell: bash
+      id: tooling-version
       if: ${{ always() && steps.go-version.outcome == 'success' }}
+      working-directory: ${{ inputs.tools-directory }}
+      run: |
+        function set-version {
+          name=$1
+          path=$2
+          version="$(
+            go mod edit --json | jq -r ".Require[] | select(.Path==\"${path}\") | .Version" || echo 'latest'
+          )"
+          echo "${name}=${version}" >> $GITHUB_OUTPUT
+        }
+
+        set-version "golangci-lint" "github.com/golangci/golangci-lint"
+        set-version "vulncheck" "golang.org/x/vuln"
+    - name: Get arguments and version for golangci-lint
+      if: ${{ always() && steps.tooling-version.outcome == 'success' }}
       id: golanci-lint
       working-directory: ${{ inputs.working-directory }}
       shell: bash
       run: |
         echo Get arguments and version for golangci-lint
         set -eu
 
-        # This handles "require foo version" and "require (\nfoo version\n)"" formats
-        version="latest"
-        if [ -n "${{ inputs.tools-directory }}" ]; then
-          modVersion="$(grep golangci-lint ${{ inputs.tools-directory }}/go.mod | rev | cut -f1 -d' ' | rev || true)"
-          [ -n "${modVersion}" ] && version=${modVersion}
-        fi
-        echo "version=${version}" >> $GITHUB_OUTPUT
-
         # Detect which config file to use
-        config=${{ github.action_path }}/.golangci-lint
+        config='${{ github.action_path }}/.golangci-lint'
         if [ -f ".golangci-lint" ]; then
           config=".golangci-lint"
         fi
-        if [ -n "${{ inputs.golangci-lint-configfile }}" ]; then
+        if [ -n '${{ inputs.golangci-lint-configfile }}' ]; then
           echo "Using project local config file"
-          config="${{ inputs.golangci-lint-configfile }}"
+          config='${{ inputs.golangci-lint-configfile }}'
         fi
         args="--config ${config}"
 
         # Go version to use
         args="${args} --go ${{ steps.go-version.outputs.version }}"
 
         # Optional build tag go-tags
-        if [ -n "${{ inputs.go-tags }}" ]; then
-          args="${args} --build-tags=${{ inputs.go-tags }}"
+        tags='${{ inputs.go-tags }}'
+        if [ -n "${tags}" ]; then
+          args="${args} --build-tags=${tags}"
         fi
         echo "args=${args}" >> $GITHUB_OUTPUT
 
     - name: Workaround for golangci-lint bug
-      if: ${{ always() && steps.golanci-lint.outcome == 'success' }}
+      if: ${{ always() && steps.golanci-lint.outcome == 'success' && runner.os == 'Linux' }}
       shell: bash
       # This step is separated from the previous one because we need to be at the repository root
       run: |
         # A terrible workaround for https://github.com/golangci/golangci-lint-action/issues/135
-        sudo chmod -R +w ../../../go/
+
+        # Avoid using sudo (it is unavailable in Windows and docker)
```

Comment:
I think you can remove in Windows here with the step condition :)

Reply:
True!